server/src/controllers/course.js

```diff
@@ -0,0 +1,25 @@
+const { getcourseByCatId } = require('../database/queries');
+
+exports.getCoursesByCatId = async (req, res, next) => {
+  try {
+    const { categoryId } = req.params;
+    if (categoryId > 0) {
+      const { rows } = await getcourseByCatId(categoryId);
+      if (rows.length !== 0) {
+        res.json({ statusCode: 200, data: rows });
+      } else {
+        res.status(404).json({
+          statusCode: 404,
+          message: "Sorry There's no courses for this category..!",
+        });
```

Comment:
**and also here, you defined status to be 404 so, there is no need to send it again within json object** 
```suggestion
        res.status(404).json({
          message: "Sorry There's no courses for this category..!",
        });
```